static/src/javascripts/projects/commercial/modules/dfp/load-advert.js

```diff
@@ -11,39 +36,29 @@ export const loadAdvert = (advert: Advert): void => {
         .then(() => {
             markTime(`Commercial: Slot Ready: ${advert.id}`);
             advert.startLoading();
-            return prebid.requestBids(advert);
+            return Promise.all([
+                prebid.requestBids(advert),
+                a9.requestBids(advert),
+            ]);
         })
-        .then(() => window.googletag.display(advert.id));
+        .then(() => {
+            window.googletag.display(advert.id);
+        });
 };
 
 export const refreshAdvert = (advert: Advert): void => {
     // advert.size contains the effective size being displayed prior to refreshing
     advert.whenSlotReady
-        .then(() =>
-            prebid.requestBids(advert, prebidSlot => {
-                // We only fiddle with top-above-nav prebidSlot(s)
-                if (prebidSlot.key !== 'top-above-nav') {
-                    return [prebidSlot];
-                }
-                // For top-above-nav slots, we force the refreshed
-                // to be the same size as the first display
-                if (prebidSlot.sizes.length === 1) {
-                    // No point forcing a size, as there is already only one
-                    // possible (mobile/tablet). See prebid/slot-config.js
-                    return [prebidSlot];
-                }
-                // Prebid slots only support array sizes (no string literals).
-                if (Array.isArray(advert.size)) {
-                    return [
-                        Object.assign({}, prebidSlot, {
-                            sizes: [[advert.size[0], advert.size[1]]],
-                        }),
-                    ];
-                }
-                // No point having this prebidSlot, as advert.size is not an array
-                return [];
-            })
-        )
+        .then(() => {
+            const prepidPromise = prebid.requestBids(advert, prebidSlot =>
+                slotFlatMap(advert, prebidSlot)
+            );
+
+            const a9Promise = a9.requestBids(advert, a9Slot =>
+                slotFlatMap(advert, a9Slot)
```

Comment:
so a9slot is the same type as prebidSlot? Maybe type annotate the signature of slotFlatMap so it's clearer.

Reply:
yap. A9 will use the exact same slots for prebid thus created the more generic `HeaderBiddingSlot` type use for both. Have now type-annotate it to make it more clear